pkg/config/structure/unmarshal.go

```diff
@@ -0,0 +1,440 @@
+	return nil, fmt.Errorf("arrayNodeImpl.GetChild not implemented")
+}
+
+// ChildrenKeys returns an error because array node does not have children accessible by name
+func (n *arrayNodeImpl) ChildrenKeys() ([]string, error) {
+	return nil, fmt.Errorf("arrayNodeImpl.ChildrenKeys not implemented")
+}
+
+// Size returns number of children in the list
+func (n *arrayNodeImpl) Size() int {
+	return n.val.Len()
+}
+
+// Index returns the kth element of the list
+func (n *arrayNodeImpl) Index(k int) (node, error) {
+	// arrayNodeImpl assumes val is an Array with Len() and Index()
+	elem := n.val.Index(k)
+	if elem.Kind() == reflect.Interface {
+		elem = elem.Elem()
+	}
+	return newNode(elem)
+}
+
+// GetChild returns an error because a leaf has no children
+func (n *leafNodeImpl) GetChild(string) (node, error) {
+	return nil, fmt.Errorf("can't GetChild of a leaf node")
+}
+
+// ChildrenKeys returns an error because a leaf has no children
+func (n *leafNodeImpl) ChildrenKeys() ([]string, error) {
+	return nil, fmt.Errorf("can't get ChildrenKeys of a leaf node")
+}
+
+// GetBool returns the scalar as a bool, or an error otherwise
+func (n *leafNodeImpl) GetBool() (bool, error) {
+	if n.val.Kind() == reflect.Bool {
+		return n.val.Bool(), nil
+	} else if n.val.Kind() == reflect.String {
+		return convertToBool(n.val.String())
+	}
+	return false, newConversionError(n.val, "bool")
+}
+
+// GetInt returns the scalar as a int, or an error otherwise
+func (n *leafNodeImpl) GetInt() (int, error) {
+	switch n.val.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return int(n.val.Int()), nil
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return int(n.val.Uint()), nil
+	case reflect.Float32, reflect.Float64:
+		return int(n.val.Float()), nil
+	}
+	return 0, newConversionError(n.val, "int")
+}
+
+// GetFloat returns the scalar as a float64, or an error otherwise
+func (n *leafNodeImpl) GetFloat() (float64, error) {
+	switch n.val.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		return float64(n.val.Int()), nil
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return float64(n.val.Uint()), nil
+	case reflect.Float32, reflect.Float64:
+		return float64(n.val.Float()), nil
+	}
+	return 0, newConversionError(n.val, "float")
+}
+
+// GetString returns the scalar as a string, or an error otherwise
+func (n *leafNodeImpl) GetString() (string, error) {
+	if n.val.Kind() == reflect.String {
+		return n.val.String(), nil
+	}
+	return "", newConversionError(n.val, "string")
+}
+
+// convert a string to a bool using standard yaml constants
+func convertToBool(text string) (bool, error) {
+	lower := strings.ToLower(text)
+	if lower == "y" || lower == "yes" || lower == "on" || lower == "true" {
+		return true, nil
+	} else if lower == "n" || lower == "no" || lower == "off" || lower == "false" {
+		return false, nil
+	}
```

Comment:
The YAML/JSON go libs also support `1`/`0` for bools.

Reply:
Ah makes sense, I'll add that before merging.